### DISCUSSION TITLE
[RFC] Hide 'JUser: :_load: Unable to load user with ID: xxx' warning message

### DISCUSSION THREAD

**Discussion by praneettekdi** [joomla/joomla-cms#39202] [General]
### Steps to reproduce the issue
If a user gets deleted from the system and if that particular user had a reference in any other component, then while loading that deleted user, Joomla throws a warning message like `JUser: :_load: Unable to load user with ID: xxx` which sometimes is not required. And it somehow disturbs the UX.

### Expected result
Of course, there is a workaround using SQL queries, but by default, **there should be a way to hide such messages.** 

### Actual result
Shows warning message: `JUser: :_load: Unable to load user with ID: xxx`

**Comment by ReLater** [comment]
See discussion in https://github.com/joomla/joomla-cms/issues/20689 , please.
And resulting "fix" for Joomla 4 which is only a change of the used language string. https://github.com/joomla/joomla-cms/pull/24166

BTW: I personally still think that the alert message is a welcome thing in backend.

**Comment by parthlawate** [comment]
The changed message is good.. However as suggested in this issue https://issues.joomla.org/tracker/joomla-cms/20689 we should only show this - say if Debug is on.. As while it could be useful to administrators showing this in the frontend might not be a good idea even if the message is changed. I think the use case here is that the JUser object is being used to load things like the Usernmame/Name instead of doing direct queries. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29855">issues.joomla.org/tracker/joomla-cms/29855</a>.</sub>

**Comment by Fedik** [comment]
I think such messages should not be show, but logged instead. There also some errors in file-system classes doing similar stuff.

Showing the error for users only makes them think that the site is broken. 
Comment https://github.com/joomla/joomla-cms/issues/20689#issuecomment-395750856

**Comment by ReLater** [comment]
> BTW: I personally still think that the alert message is a welcome thing **in backend.**

Many commercial templates have that issue at least (but not only) when one deletes the pre-configured users.
